File: backend/src/main/java/com/tmk/vtcmanager/application/domain/operation/CatalogueElementMaintenance.java
package com.tmk.vtcmanager.application.domain.operation;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CatalogueElementMaintenance {

    private Long id;
    private String libelle;
    private boolean actif;

    /** Montant pré-rempli à la saisie d'une maintenance (facultatif). */
    private BigDecimal montantDefaut;

    /** Nom d'objet de l'image d'illustration dans le stockage (facultatif). */
    private String image;

    public void update(String libelle, BigDecimal montantDefaut, String image) {
        this.libelle = libelle;
        this.montantDefaut = montantDefaut;
        this.image = image;
    }

    public void changerActivation(boolean actif) {
        this.actif = actif;
    }
}

File: mobile/lib/features/operation_financiere/domain/entities/catalogue_element_maintenance.dart
class CatalogueElementMaintenance {
  final int id;
  final String libelle;
  final bool actif;

  /// Montant pré-rempli à la sélection dans une opération de maintenance.
  final double? montantDefaut;

  /// URL présignée de l'image d'illustration (facultative).
  final String? imageUrl;

  const CatalogueElementMaintenance({
    required this.id,
    required this.libelle,
    required this.actif,
    this.montantDefaut,
    this.imageUrl,
  });
}
